lib/support/clone_vm.rb

```diff
@@ -272,38 +288,79 @@ def active_ip_discovery(prefix_commands = [])
       return unless active_discovery? || instant_clone?
 
       Kitchen.logger.info "Attempting active IP discovery"
-      begin
-        tools = Support::GuestOperations.new(vim, vm, guest_auth, ssl_verify)
-
-        commands = []
-        commands << rescan_commands if instant_clone?
-        # commands << trigger_tools # deactivated for now, as benefit is doubtful
-        commands << discovery_commands
-        script = commands.flatten.join(command_separator)
-
-        stdout = tools.run_shell_capture_output(script, :auto, 20)
-
-        # Windows returns wrongly encoded UTF-8 for some reason
-        stdout = stdout.bytes.map { |b| (32..126).cover?(b.ord) ? b.chr : nil }.join unless stdout.ascii_only?
-        @ip = stdout.match(/([0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3})/m)&.captures&.first
-
-        Kitchen.logger.debug format("Script output: %s", stdout)
-        raise Support::CloneError.new(format("Could not find IP in script output, fallback to standard discovery")) if ip.nil?
-        raise Support::CloneError.new(format("Error getting accessible IP address, got %s. Check DHCP server, scope exhaustion or timing issues", ip)) if ip =~ /^169\.254\./
-      rescue RbVmomi::Fault => e
-        if e.fault.class.wsdl_name == "InvalidGuestLogin"
-          message = format('Error authenticating to guest OS as "%s", check configuration of "vm_username"/"vm_password"', username)
-        else
-          message = e.message
-        end
 
-        raise Support::CloneError.new(message)
-      rescue ::StandardError => e
-        Kitchen.logger.info format("Active discovery failed: %s", e.message)
-        return false
-      end
+      commands = []
+      commands << rescan_commands if instant_clone?
+      # commands << trigger_tools # deactivated for now, as benefit is doubtful
+      commands << discovery_commands
+      script = commands.flatten.join(command_separator)
+
+      output = execute_via_tools(script)
+
+      @ip = output.match(/([0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3})/m)&.captures&.first
+      raise Support::CloneError.new(format("Could not find IP in script output, fallback to standard discovery")) if ip.nil?
+      raise Support::CloneError.new(format("Error getting accessible IP address, got %s. Check DHCP server, scope exhaustion or timing issues", ip)) if ip =~ /^169\.254\./
 
       true
+
+    rescue ::StandardError => e
+      Kitchen.logger.info format("Active discovery failed: %s", e.message)
+
+      false
+    end
+
+    def execute_via_tools(script, timeout = 20)
+      Kitchen.logger.debug format("Script execution: %s", script)
+
+      tools = Support::GuestOperations.new(vim, vm, guest_auth, ssl_verify)
+      output = tools.run_shell_capture_output(script, :auto, timeout)
```

Comment:
Hrm, when I run this with a Windows 2016 template with VMWare tools installed, this fails at this point because it fails to write a temporary file:

```
-----> Creating <default-windows-2016>...
       Assigning network VM Network...
       Cloning 'Templates/windows2016-template' to create the VM...
       Waiting for VMware tools to become available (timeout: 90 seconds)...
       Running post create script 'kitchen.yml'
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: 1 actions failed.
>>>>>>     Failed to complete #create action: [FileNotFound: File C:\Windows\TEMP\vm_utils_run_out_0.2658098154379308 was not found] on default-windows-2016
>>>>>> ----------------------
>>>>>> Please see .kitchen/logs/kitchen.log for more details
>>>>>> Also try running `kitchen diagnose --all` for configuration
```

My Kitchenfile:
```
driver:
  name: vcenter
  vcenter_username: (...)
  vcenter_password: (...)
  vcenter_host:  (...)

platforms:
  - name: windows-2016
    driver:
      targethost: (...)
      network_name: VM Network
      folder: cwolfe
      clone_type: full
      vm_wait_timeout: 180
      datacenter: (...)
      template: Templates/windows2016-template
      post_create_script: |
        netsh advfirewall firewall set rule name="Windows Remote Management (HTTP-In)" profile="Public" new remoteip="any"
    transport:
      username: vagrant
      password: vagrant

suites:
  - name: default

provisioner:
  name: chef_zero

verifier:
  name: inspec
```
`C:\Windows\Temp\` exists. Any advice?